### PR TITLE
Feature add db support for language translation

### LINKED ISSUE #1651: Dynamic strings
- [x] Add a DB table to manage strings & translations
- [x] Strings for options & questions will come from the local DB instead of Android default system
- [ ] Create CSV for strings and populate DB

### PR DESCRIPTION
**Pull request by Amejia481**
Close #1651

### CODE REVIEW

File: app/src/main/java/org/eyeseetea/malariacare/utils/Utils.java [hunk 1]
```diff
@@ -58,44 +62,70 @@ public static String round(float base) {
...
+
+    @NonNull
+    public static String getInternationalizedString(@NonNull String key) {
+        Context ctx = PreferencesState.getInstance().getContext();
+        String language = getUserLanguage(ctx);
+        String translation = getLocalizedStringFromDB(key, language);
+
+        if (key == null) {
+            return key;
+        }
```
Comment (ifoche) [inline app/src/main/java/org/eyeseetea/malariacare/utils/Utils.java]: @Amejia481 on null we were previously returning an empty string, and here you changed that behavior into return null. Is there any special reason for doing that? Did you check which consequences that could have? I would prefer to return anything but a null if possible, unless there's a good reason for not doing it

**Review by ifoche — CHANGES_REQUESTED** [review]
@Amejia481 only a very minor comment. If you can check that returned null consequences that would be great. I will not merge this code until Login changes are applied to #1670 as this code comes from that branch. I have really appreciated the indexed solution for the strings. It's been really smart. Thank you for that.

File: app/src/main/java/org/eyeseetea/malariacare/utils/Utils.java (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/eyeseetea/malariacare/utils/Utils.java]: You're right, my bad, It looks like I missed this one.

Comment (Amejia481) [reply app/src/main/java/org/eyeseetea/malariacare/utils/Utils.java]: @idelcano I updated the PR with the change, thanks for catching that one!

**Review by ifoche — APPROVED** [review]
thanks @Amejia481

**Pull request merged**